src/codeu/chat/util/TransactionLogger.java

```diff
@@ -0,0 +1,155 @@
+
+import java.io.*;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.LinkedList;
+import java.lang.System;
+
+import codeu.chat.common.User;
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import com.google.gson.JsonObject;
+
+
+public class TransactionLogger {
+
+	private LinkedList<String> toLog; //Implements Queue
+	private long lastWrite;
+
+	public TransactionLogger() {
+		toLog = new LinkedList<String>();
+		lastWrite = System.currentTimeMillis();
+	}
+
+  // methods to serialize commands.
+  public void addUser(User user) {
+    UserJson userJson = new UserJson("ADD-USER", user);
+    Gson gson = new Gson();
+    log(gson.toJson(userJson));
+  }
+
+  public void addConversation(ConversationHeader conversation) {
+    ConversationJson conversationJson = new ConversationJson("ADD-CONVERSATION", conversation);
+    Gson gson = new Gson();
+    log(gson.toJson(conversationJson));
+  }
+
+  public void addMessage(Uuid conversation, Message message) {
+    MessageJson messageJson = new MessageJson("ADD-MESSAGE", conversation, message);
+    Gson gson = new Gson();
+    log(gson.toJson(messageJson));  
+  }
+
+    // methods to deserialze commands.
+    public void retrieveUser(Controller controller, JsonObject jsonObject){
+        try {
+            Uuid id = Uuid.parse(jsonObject.get("uuid").getAsString());
+            String name = jsonObject.get("name").getAsString();
+            SimpleDateFormat formatter = new SimpleDateFormat("dd-MMM-yyyy HH:mm:ss.SSS");
+            Time creation = Time.fromMs(formatter.parse(jsonObject.get("creation").getAsString()).getTime());
+            controller.newUser(id, name, creation);
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void retrieveConversation(Controller controller, JsonObject jsonObject){
+        try {
+            Uuid id = Uuid.parse(jsonObject.get("uuid").getAsString());
+            Uuid owner = Uuid.parse(jsonObject.get("owner").getAsString());
+            SimpleDateFormat formatter = new SimpleDateFormat("dd-MMM-yyyy HH:mm:ss.SSS");
+            Time creation = Time.fromMs(formatter.parse(jsonObject.get("creation").getAsString()).getTime());
+            String title = jsonObject.get("title").getAsString();
+            controller.newConversation(id, title, owner, creation);
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void retrieveMessage(Controller controller, JsonObject jsonObject){
+        try {
+            Uuid id = Uuid.parse(jsonObject.get("uuid").getAsString());
+            Uuid conversation = Uuid.parse(jsonObject.get("conversation").getAsString());
+            SimpleDateFormat formatter = new SimpleDateFormat("dd-MMM-yyyy HH:mm:ss.SSS");
```

Comment:
Nit: So there's a gson.from(String, Class) method that would make life a little easier here